internal/services/eureka/eureka_resolver.go

```diff
@@ -0,0 +1,101 @@
+/*
+ *
+ * Copyright 2018 gRPC authors.
+ * Copyright 2020 Justin Haygood
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package eureka
+
+import (
+	"log"
+	"strconv"
+	"strings"
+
+	ec "github.com/ArthurHlt/go-eureka-client/eureka"
+	"google.golang.org/grpc/resolver"
+)
+
+// NewEurekaBuilder creates a eurekaBuilder which is used to factory DNS resolvers.
+func NewEurekaBuilder() resolver.Builder {
+	return &eurekaBuilder{}
+}
+
+type eurekaBuilder struct{}
+
+// Build creates and starts a DNS resolver that watches the name resolution of the target.
+func (b *eurekaBuilder) Build(target resolver.Target, cc resolver.ClientConn, opts resolver.BuildOptions) (resolver.Resolver, error) {
+
+	eurekaServer := target.Authority
+	serviceName := target.Endpoint
+
+	if len(serviceName) == 0 {
+		serviceName = eurekaServer
+		eurekaServer = "localhost:8761/eureka"
+	}
+
+	serviceNameIndex := strings.LastIndex(serviceName, "/")
+
+	if serviceNameIndex != -1 {
+		eurekaServerPath := serviceName[0:serviceNameIndex]
+		serviceName = serviceName[serviceNameIndex+1:]
+		eurekaServer = eurekaServer + "/" + eurekaServerPath
+	}
+
+	d := &eurekaResolver{EurekaServer: eurekaServer, ServiceName: serviceName, ClientConn: cc}
+
+	d.ResolveNow(resolver.ResolveNowOptions{})
+
+	return d, nil
+}
+
+// Scheme returns the naming scheme of this resolver builder, which is "dns".
+func (b *eurekaBuilder) Scheme() string {
+	return "eureka"
+}
+
+type eurekaResolver struct {
+	EurekaServer string
+	ServiceName  string
+	ClientConn   resolver.ClientConn
+}
+
+// ResolveNow invoke an immediate resolution of the target that this dnsResolver watches.
+func (d *eurekaResolver) ResolveNow(resolver.ResolveNowOptions) {
+	eurekaClient := ec.NewClient([]string{
+		"http://" + d.EurekaServer,
+	})
+
+	application, err := eurekaClient.GetApplication(d.ServiceName)
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var newAddrs []resolver.Address = make([]resolver.Address, 0, len(application.Instances))
+
+	for _, instance := range application.Instances {
+		addr := instance.IpAddr + ":" + strconv.Itoa(application.Instances[0].Port.Port)
```

Comment:
Maybe we should create an issue and use the uri? Its probably throughout the whole project

https://golang.org/pkg/net/url/

Reply:
The resolver.Address is supposed to be ip:port, not a full URI, and I verified that by using the default DNS resolver as reference.

Reply:
I guess I'm really referring to `"http://" + d.EurekaServer`. 

I'm just wondering if we see a value in switching to URI's where appropriate or leave as is. I can open an issue instead and we can decide latter if we should even change it.

Reply:
So something like this?

```go
eurekaUrl := url.Url{Schema="http",Host=d.EurekaServer,Path=d.EurekaPath}

eurekaClient := ec.NewClient([]string{
		eurekaUrl.String(),
	})
```

Reply:
Changed to use net/url